index.js

```diff
@@ -1,109 +1,201 @@
+  'eth_getTransactionByBlockHashAndIndex',
+);
+EthQuery.prototype.getTransactionByBlockNumberAndIndex = generateFnFor(
+  'eth_getTransactionByBlockNumberAndIndex',
+);
+EthQuery.prototype.getTransactionReceipt = generateFnFor(
+  'eth_getTransactionReceipt',
+);
+EthQuery.prototype.getUncleByBlockHashAndIndex = generateFnFor(
+  'eth_getUncleByBlockHashAndIndex',
+);
+EthQuery.prototype.getUncleByBlockNumberAndIndex = generateFnFor(
+  'eth_getUncleByBlockNumberAndIndex',
+);
+EthQuery.prototype.getCompilers = generateFnFor('eth_getCompilers');
+EthQuery.prototype.compileLLL = generateFnFor('eth_compileLLL');
+EthQuery.prototype.compileSolidity = generateFnFor('eth_compileSolidity');
+EthQuery.prototype.compileSerpent = generateFnFor('eth_compileSerpent');
+EthQuery.prototype.newFilter = generateFnFor('eth_newFilter');
+EthQuery.prototype.newBlockFilter = generateFnFor('eth_newBlockFilter');
+EthQuery.prototype.newPendingTransactionFilter = generateFnFor(
+  'eth_newPendingTransactionFilter',
+);
+EthQuery.prototype.uninstallFilter = generateFnFor('eth_uninstallFilter');
+EthQuery.prototype.getFilterChanges = generateFnFor('eth_getFilterChanges');
+EthQuery.prototype.getFilterLogs = generateFnFor('eth_getFilterLogs');
+EthQuery.prototype.getLogs = generateFnFor('eth_getLogs');
+EthQuery.prototype.getWork = generateFnFor('eth_getWork');
+EthQuery.prototype.submitWork = generateFnFor('eth_submitWork');
+EthQuery.prototype.submitHashrate = generateFnFor('eth_submitHashrate');
 
 // network level
 
-EthQuery.prototype.sendAsync = function(opts, cb){
-  const self = this
-  self.currentProvider.sendAsync(createPayload(opts), function(err, response){
-    if (!err && response.error) err = new Error('EthQuery - RPC Error - '+response.error.message)
-    if (err) return cb(err)
-    cb(null, response.result)
-  })
-}
+EthQuery.prototype.sendAsync = function (opts, callback) {
+  const self = this;
+  self.currentProvider.sendAsync(
+    createPayload(opts),
+    function (error, response) {
+      let improvedError = error;
+      if (!error && response.error) {
+        improvedError = new Error(
+          `EthQuery - RPC Error - ${response.error.message}`,
+        );
+      }
+      if (improvedError) {
+        return callback(improvedError);
+      }
+      return callback(null, response.result);
+    },
+  );
+};
 
 // util
 
-function generateFnFor(methodName){
-  return function(){
-    const self = this
-    var args = [].slice.call(arguments)
-    var cb = args.pop()
-    self.sendAsync({
-      method: methodName,
-      params: args,
-    }, cb)
-  }
+/**
+ * Generates an instance method designed to call an RPC method that takes no
+ * parameters. This instance method uses `sendAsync` internally to make the
+ * request to the network.
+ *
+ * @param {any} methodName - The RPC method.
+ * @returns {any} The method.
+ */
+function generateFnFor(methodName) {
+  return function (...args) {
```

Comment:
I had to change this function along with the next function slightly to satisfy the linter, as `arguments` is disallowed. This is of course modern syntax, but it will be transpiled away when I convert this codebase to TypeScript.